tests/core/v5_1/test_network.py

```diff
@@ -100,6 +101,93 @@ async def test_network_find_nodes_api(alice, bob):
     assert response_distances == {256, 255}
 
 
+@pytest.fixture
+async def bob_client(bob):
+    async with bob.client() as bob_client:
+        yield bob_client
+
+
+@pytest.fixture
+async def alice_network(alice):
+    async with alice.network() as alice_network:
+        yield alice_network
+
+
+@pytest.mark.trio
+async def test_network_lookup_empty_response(bob, alice_network, alice, bob_client):
+    async def return_empty_response():
+        async with bob.events.find_nodes_received.subscribe() as subscription:
+            find_nodes = await subscription.receive()
+
+            await bob_client.send_found_nodes(
+                alice.node_id,
+                alice.endpoint,
+                enrs=[],
+                request_id=find_nodes.message.request_id,
+            )
+
+    with trio.fail_after(2):
+        async with trio.open_nursery() as nursery:
+            nursery.start_soon(return_empty_response)
+
+            with pytest.raises(Exception):
+                await alice_network.lookup_enr(bob.node_id, enr_seq=101)
+
+
+@pytest.mark.trio
+async def test_network_lookup_normal_response(bob, alice_network, alice, bob_client):
+    async def return_normal_response():
+        async with bob.events.find_nodes_received.subscribe() as subscription:
+            find_nodes = await subscription.receive()
+
+            await bob_client.send_found_nodes(
+                alice.node_id,
+                alice.endpoint,
+                enrs=[bob.enr],
+                request_id=find_nodes.message.request_id,
+            )
+
+    with trio.fail_after(2):
+        async with trio.open_nursery() as nursery:
+            nursery.start_soon(return_normal_response)
+
+            enr = await alice_network.lookup_enr(bob.node_id, enr_seq=101)
+            assert enr == bob.enr
+
+
+@pytest.mark.trio
+async def test_network_lookup_many_enr_response(bob, alice_network, alice, bob_client):
+
+    enr_manager = ENRManager(private_key=bob.private_key, enr_db=bob.enr_db,)
+    enr_manager.update(
+        (b"udp", bob.endpoint.port), (b"ip", bob.endpoint.ip_address),
+    )
+
+    first_enr = enr_manager.enr
+
+    enr_manager.update((b"eth2", b"\x00\x00"))
+
+    second_enr = enr_manager.enr
```

Comment:
You might make an assertion that you have two different sequence numbers here just for the sake of sanity checking.